File: Graphs/UndirectedGraphs/examples/undirected_graph_example.py
from Graphs.UndirectedGraphs.source.undirected_graph import UndirectedGraph


if __name__ == "__main__":
    # Create an undirected graph
    graph = UndirectedGraph()

    # Add vertices to the graph
    graph.add_vertex("A")
    graph.add_vertex("B")
    graph.add_vertex("C")
    graph.add_vertex("D")

    # Add edges between vertices
    graph.add_edge("A", "B")  # A -- B
    graph.add_edge("A", "C")  # A -- C
    graph.add_edge("B", "D")  # B -- D
    graph.add_edge("C", "D")  # C -- D

    # Print the graph\'s adjacency list
    print("Graph Adjacency List:")
    graph.print()

    # Check if specific edges exist
    print("\nChecking if edges exist:")
    print(f"Does an edge exist between A and B? {'Yes' if graph.has_edge('A', 'B') else 'No'}")
    print(f"Does an edge exist between A and D? {'Yes' if graph.has_edge('A', 'D') else 'No'}")

File: Graphs/UndirectedGraphs/examples/undirected_graph_example.js
const UndirectedGraph = require('../source/undirected_graph');

// Create an undirected graph
const graph = new UndirectedGraph();

// Add vertices to the graph
graph.addVertex('A');
graph.addVertex('B');
graph.addVertex('C');
graph.addVertex('D');

// Add edges between vertices
graph.addEdge('A', 'B');  // A -- B
graph.addEdge('A', 'C');  // A -- C
graph.addEdge('B', 'D');  // B -- D
graph.addEdge('C', 'D');  // C -- D

// Print the graph's adjacency list
console.log('Graph Adjacency List:');
graph.print();

// Check if specific edges exist
console.log('\nChecking if edges exist:');
console.log(`Does an edge exist between A and B? ${graph.hasEdge('A', 'B') ? 'Yes' : 'No'}`);
console.log(`Does an edge exist between A and D? ${graph.hasEdge('A', 'D') ? 'Yes' : 'No'}`);
